api/webhooks/webhooks.py

```diff
@@ -43,31 +47,45 @@ class WebhookType(enum.Enum):
 }
 
 
-def get_webhook_model(webhook_type: WebhookType) -> typing.Union[WebhookModels]:
+def get_webhook_model(
+    webhook_type: WebhookType,
+) -> Type[Union[OrganisationWebhook, Webhook]]:
     if webhook_type == WebhookType.ORGANISATION:
         return OrganisationWebhook
     if webhook_type == WebhookType.ENVIRONMENT:
         return Webhook
 
 
-def call_environment_webhooks(environment, data, event_type):
+@register_task_handler()
+def call_environment_webhooks(
+    environment_id: int, data: typing.Mapping, event_type: str
+):
     if settings.DISABLE_WEBHOOKS:
         return
-
+    try:
+        environment = Environment.objects.get(id=environment_id)
+    except Environment.DoesNotExist:
+        return
     _call_webhooks(
-        environment.webhooks.filter(enabled=True),
+        (webhook.id for webhook in environment.webhooks.filter(enabled=True)),
```

Comment:
I don't think we need to worry about the serialization here, right? Once we're at this point, everything is contained within the run time of the task processor. We can pass any objects between functions at this point.

Reply:
Yes, true. We don't need to serialize the arguments to `_call_webhooks`, but `_call_webhooks` calls `_call_webhook_email_on_error.delay`, which should be async so that unhealthy webhooks don't block the for loop. So even if I pass the `webhook` object completely, I would still have to get the serializable `webhook.id` to schedule `_call_webhook_email_on_error`. 


```python
def _call_webhooks(
    webhook_ids: typing.Iterable[int],
    data: typing.Mapping,
    event_type: str,
    webhook_type: WebhookType,
):
    webhook_data = {"event_type": event_type, "data": data}
    serializer = WebhookSerializer(data=webhook_data)
    serializer.is_valid(raise_exception=False)
    for webhook_id in webhook_ids:
        _call_webhook_email_on_error.delay(
            args=(webhook_id, serializer.data, webhook_type.value)
        )
```

I can also get the `webhook.id` in the `_call_webhooks` function, let me know what you think.

Reply:
Yes, I think we should just change the call to `_call_webook_email_on_error.delay` to pass `webhook.id` and pass in the webhook objects to `_call_webhooks`.

Reply:
Sure :)